### PR TITLE
LPS-183396 Enable account restriction for Proxy Salesforce object

### PR DESCRIPTION
**Pull request by TamyrisTorres**

### CODE REVIEW

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java [hunk 1]
```diff
@@ -4355,6 +4331,53 @@ private void _associateUserAccounts(
 		}
 	}
 
+	private void _enableRestrictedObject(ServiceContext serviceContext)
+		throws Exception {
+
+		String json = SiteInitializerUtil.read(
+			"/site-initializer/restrict-account-entry.json",
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java:4338]: refactor to work as object actions

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java [hunk 2]
```diff
@@ -4355,6 +4331,53 @@ private void _associateUserAccounts(
...
+		if (json == null) {
+			return;
+		}
+
+		JSONArray jsonArray = _jsonFactory.createJSONArray(json);
+
+		for (int i = 0; i < jsonArray.length(); i++) {
+			JSONObject jsonObject = jsonArray.getJSONObject(i);
+
+			com.liferay.object.model.ObjectDefinition objectDefinition2 =
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java:4350]: rename to serviceBuilderObjectDefinition

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java [hunk 3]
```diff
@@ -490,29 +490,34 @@ public void deployObjectDefinition(ObjectDefinition objectDefinition) {
 	}
 
 	@Override
-	public ObjectDefinition enableAccountEntryRestricted(
-			ObjectRelationship objectRelationship)
+	public ObjectDefinition enableObjectRestricted(
```
Comment (nNilton) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java:493]: create another method to set account restriction for salesforce

Comment (nNilton) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java:493]: you could rename to enableSalesForceAccountEntryRestricted and enableAccountEntryRestricted

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test-bundle-2/src/main/resources/site-initializer/restrict-account-entry.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+[
+	{
+		"objectDefinitionName": "TestObjectDefinition4",
+		"relationShipName": "accountEntryToTestObjectDefinition4"
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test-bundle-2/src/main/resources/site-initializer/restrict-account-entry.json:4]: rename to objectRelationshipName

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test-bundle-2/src/main/resources/site-initializer/restrict-account-entry.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+	{
+		"objectDefinitionName": "TestObjectDefinition5",
+		"relationShipName": "accountEntryToTestObjectDefinition5"
+
+	},
+	{
+		"objectDefinitionName": "TestObjectDefinition1",
+		"objectField": "field1"
+	}
+]
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test-bundle-2/src/main/resources/site-initializer/restrict-account-entry.json:16]: sort

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectDefinitionLocalServiceTest.java [hunk 6]
```diff
@@ -1051,7 +1051,7 @@ public void testEnableAccountEntryRestricted() throws Exception {
 				ObjectRelationshipConstants.TYPE_ONE_TO_MANY);
 
 		objectDefinition2 =
-			_objectDefinitionLocalService.enableAccountEntryRestricted(
+			_objectDefinitionLocalService.enableObjectRestricted(
 				objectRelationship);
 
 		Assert.assertTrue(objectDefinition2.isAccountEntryRestricted());
```
Comment (nNilton) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectDefinitionLocalServiceTest.java:1057]: add assert to check accountRestrictionFieldId

**Comment by TamyrisTorres** [comment]
https://github.com/pt-liferay-solutions/liferay-portal/pull/4063

**Pull request closed without merging**